### PR TITLE
Add Data Synthesis cookbook.

### PR DESCRIPTION
**Pull request by prescod**
Add a data synthesis cookbook to the documentation.

### CODE REVIEW

File: docs/cookbook.rst [hunk 1]
```diff
@@ -534,3 +534,107 @@ References
...
+then when you run the command above it will run the recipe
+100 times (100*10=1000) which will generate 1000 Accounts, 500 Contacts
+and 1500 Opportunites.
+
+Batch Sizes
+-----------
+
+You can also control batch sizes with the ``-o batch_size BATCHSIZE``
+parameter. This is not the Salesforce bulk API batch size. No matter
+what batch size you select, CumulusCI wll properly split your data
```
Comment (davisagli) [inline docs/cookbook.rst:589]:
```suggestion
what batch size you select, CumulusCI will properly split your data
```

File: docs/cookbook.rst [hunk 2]
```diff
@@ -534,3 +534,107 @@ References
...
+into 10k batches for the bulk API.
```
Comment (davisagli) [inline docs/cookbook.rst:590]:
```suggestion
into batches for the bulk API.
```

(there is a record count limit and a byte size limit)

**Review by davisagli — CHANGES_REQUESTED** [review]

**Review by davisagli — DISMISSED** [review]

File: docs/cookbook.rst [hunk 3]
```diff
@@ -534,3 +534,107 @@ References
...
+100 times (100*10=1000) which will generate 1000 Accounts, 500 Contacts
+and 1500 Opportunites.
+
+Batch Sizes
+-----------
+
+You can also control batch sizes with the ``-o batch_size BATCHSIZE``
+parameter. This is not the Salesforce bulk API batch size. No matter
+what batch size you select, CumulusCI will properly split your data
+into batches for the bulk API.
```
Comment (Br4nd0R) [inline docs/cookbook.rst:590]: 🎆

File: docs/cookbook.rst [hunk 4]
```diff
@@ -534,3 +534,107 @@ References
...
+uploads are always done in batches of records a particular SObject.
+So in the case above, it would upload 1000 Accounts, then 500
+Contacts, then 1500 Opportunites. (remember that our scenario
+involves a recipe that generates 10 Accounts, 5 Contacts and 15
+Opportunites).
+
+Imagine if the numbers were more like 1M, 500K and 1.5M. And further,
+imagine if your network crashed after 1M Accounts and 499K Contacts 
+were uploaded. You would not have a single "complete set" of 10/5/15.
+Instad you would have 1M "partial sets".
```
Comment (Br4nd0R) [inline docs/cookbook.rst:614]:
```suggestion
Instead you would have 1M "partial sets".
```

**Review by davisagli — APPROVED** [review]

**Pull request merged**